module-5/src/lib/alchemy.ts

```diff
@@ -0,0 +1,82 @@
+import { Alchemy, Network, AssetTransfersCategory, Block } from "alchemy-sdk";  
+
+// Configure Alchemy SDK  
+if (!process.env.NEXT_PUBLIC_ALCHEMY_API_KEY || 
+    process.env.NEXT_PUBLIC_ALCHEMY_API_KEY === 'demo') {
+  throw new Error('Please set NEXT_PUBLIC_ALCHEMY_API_KEY in .env.local');
+}
+
+const settings = {  
+  apiKey: process.env.NEXT_PUBLIC_ALCHEMY_API_KEY,
+  network: Network.ETH_MAINNET,  
+};  
+
+export const alchemy = new Alchemy(settings);
+
+// Type for enriched block data  
+export interface EnrichedBlock {  
+  number: number;  
+  timestamp: number;  
+  gasUsed: number;  
+  gasLimit: number;  
+  baseFeeGwei: number;  
+  gasRatio: number;  
+  transferVolume: number;  
+  transactions: number;  
+}  
+
+// Fetch block data with additional metrics  
+export const getEnrichedBlock = async (blockNumber: number, tokenAddress: string): Promise<EnrichedBlock> => {  
+  // Log input parameters
+  
+  console.log('Getting block data:', {
+    blockNumber,
+    tokenAddress,
+    hexBlockNumber: "0x" + blockNumber.toString(16)
+  });
+
+  const block = await alchemy.core.getBlock(blockNumber);
+  
+  // Validate token address
+  if (!tokenAddress) {
+    console.error('Token address is null or undefined');
+    throw new Error('Invalid token address');
+  }
+
+  // Log transfer request parameters
+  const transferParams = {
+    fromBlock: "0x" + blockNumber.toString(16),
+    toBlock: "0x" + blockNumber.toString(16),
+    contractAddresses: [tokenAddress],
+    category: [AssetTransfersCategory.ERC20],
+  };
+
+  try {
+    const transfers = await alchemy.core.getAssetTransfers(transferParams);
```

Comment:
- [ ] in this assignment you are required to filter transfer events from ERC20 Transfer event using logs : Refer : https://www.alchemy.com/docs/deep-dive-into-eth_getlogs